homeassistant/components/openai_conversation/__init__.py

```diff
@@ -0,0 +1,138 @@
+"""The OpenAI Conversation integration."""
+from __future__ import annotations
+
+from functools import partial
+import logging
+from typing import cast
+
+import openai
+from openai import error
+
+from homeassistant.components import conversation
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, TemplateError
+from homeassistant.helpers import area_registry, device_registry, intent, template
+from homeassistant.util import ulid
+
+from .const import DEFAULT_MODEL, DEFAULT_PROMPT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up OpenAI Conversation from a config entry."""
+    openai.api_key = entry.data[CONF_API_KEY]
```

Comment:
This is the only way one can configure an API key in OpenAI and it's dumb, but sadly the way it is.